cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/YBSnapshotWithPgSinkIT.java

```diff
@@ -0,0 +1,323 @@
+        int rowsToBeInserted = 100;
+        // Insert data to the source table
+        TestHelper.insertRowsInSourceTable(ybHelper, DEFAULT_TABLE_NAME, rowsToBeInserted);
+
+        // Start the CDCSDKContainer with initial snapshot mode
+        cdcsdkContainer = kafkaHelper.getCdcsdkContainer(ybHelper, "public." + DEFAULT_TABLE_NAME, 10, "initial");
+        cdcsdkContainer.withNetwork(containerNetwork);
+        cdcsdkContainer.start();
+
+        pgHelper.waitTillRecordsAreVerified(rowsToBeInserted, 10000);
+
+        // Now stop the cdcsdk container and start it again with never snapshot mode
+        cdcsdkContainer.stop();
+
+        // Change the snapshot mode and start the container
+        cdcsdkContainer.withEnv("CDCSDK_SOURCE_SNAPSHOT_MODE", "never");
+        cdcsdkContainer.start();
+
+        for (int i = 100; i < 150; ++i) {
+            ybHelper.execute(UtilStrings.getInsertStmt(DEFAULT_TABLE_NAME, i, "first_" + i, "last_" + i, 23.45));
+        }
+
+        pgHelper.waitTillRecordsAreVerified(150, 10000);
+        pgHelper.assertRecordCountInPostgres(150);
+
+        ybHelper.execute("UPDATE " + DEFAULT_TABLE_NAME + " SET first_name = 'Vaibhav' WHERE id >= 100 AND id <= 149;");
+
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(5))
+                .until(() -> {
+                    ResultSet rSet = pgHelper.executeAndGetResultSet("SELECT COUNT(*) FROM test_table WHERE first_name='Vaibhav';");
+                    rSet.next();
+                    return rSet.getInt(1) == 50;
+                });
+
+        // Check the results in Postgres
+        ResultSet rs = pgHelper.executeAndGetResultSet(String.format("SELECT * FROM %s WHERE id >= 100 and id <= 149 ORDER BY id;", DEFAULT_TABLE_NAME));
+        int ind = 100;
+        while (rs.next()) {
+            TestHelper.assertValuesInResultSet(rs, ind, "Vaibhav", "last_" + ind, 23.45);
+            ++ind;
+        }
+
+        ybHelper.execute("DELETE FROM " + DEFAULT_TABLE_NAME + ";");
+
+        // Wait till all the records are deleted from the sink
+        pgHelper.waitTillRecordsAreVerified(0, 10000);
+    }
+
+    @Test
+    public void restartCdcsdkServerWhileSnapshotInProgress() throws Exception {
+        int rowsToBeInserted = 500;
+        TestHelper.insertRowsInSourceTable(ybHelper, DEFAULT_TABLE_NAME, rowsToBeInserted);
+
+        // Get the cdcsdk container instance
+        cdcsdkContainer = kafkaHelper.getCdcsdkContainer(ybHelper, "public." + DEFAULT_TABLE_NAME, 10, "initial");
+        cdcsdkContainer.withNetwork(containerNetwork);
+        cdcsdkContainer.start();
+
+        // Wait for at least 1000 rows to be replicated across postgres sink and then restart the cdcsdk server
+        // Doing the above would effectively mean that the server will take the snapshot again
+        // Ignore the exceptions of type SQLException which may be thrown if the table does not exist
+        Awaitility.await()
```

Comment:
How do we know if all the data has not be streamed here? It could be in 1 or 2 batche.s

Reply:
I will increase the number of rows to be inserted so that when we reach this point in the code flow then we could be sure that not all the records have been streamed.